### PR TITLE
Feature/change feed

### PR DESCRIPTION
**Pull request by rickykaare**
Add support for adding change feed processors

### CODE REVIEW

File: src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        public CosmosBuilder(
+            IServiceCollection services)
+            : base(services)
+        {
+        }
+
+        public ICosmosBuilder<T> WithChangeFeedProcessor<TProcessor>()
+            where TProcessor : class, IChangeFeedProcessor<T>
+        {
+            Services.TryAddSingleton<ICosmosContainerInitializer, LeasesContainerInitializer>();
```
Comment (rickykaare) [inline src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs:19]: This probably needs to be added even if another ICosmosInitializer has been added.

Comment (rickykaare) [reply src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs:19]: Will fix

File: src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            Services.TryAddSingleton<IChangeFeedFactory, ChangeFeedFactory>();
+            Services.AddSingleton<TProcessor>();
+
+            Services.AddHostedService<ChangeFeedService<T, TProcessor>>();
```
Comment (LarsSkovslund) [inline src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs:23]: How do you ensure  that all `ICosmosContainerInitializer`'s are completed before the hosted service is started?

> If they don't the hosted service will fail and never start.

**Review by LarsSkovslund — CHANGES_REQUESTED** [review]

File: src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs (hunk 2 above)
Comment (rickykaare) [reply src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs:23]: Good point. Will fix

**Review by LarsSkovslund — APPROVED** [review]

**Pull request merged**